syntax = "proto3";
package helloworld;

service Hello {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloTwice (HelloRequest) returns (stream HelloReply) {}
  rpc SayHelloRepeatedly (stream HelloRequest) returns (stream HelloReply) {}
  rpc SayHelloLast (stream HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
